src/Routines.ts

```diff
@@ -169,37 +190,48 @@ export module Routines {
             const msgAdd: string = (toAdd && toAdd.length) ? `Adding: [${toAdd.map(d => manager.dict[d].name).join(", ")}].` : "";
             const choice: string = await vscode.window.showWarningMessage(`${msgRemove} ${msgAdd} Proceed?`, "Proceed", "Cancel");
             if (choice !== "Proceed") {
+                TelemetryHelper.finishStep(stepCancel);
                 return;
+            } else {
+                TelemetryHelper.finishStep(stepProceed);
             }
+
             // modify xml object
-            const newXml: any = Object.assign({}, xml);
+            const newXml: { project: XmlNode } = getUpdatedPomXml(xml, starters, toRemove, toAdd);
+
+            // re-generate a pom.xml
+            const output: string = Utils.buildXmlContent(newXml);
+            await fse.writeFile(entry.fsPath, output);
+            vscode.window.showInformationMessage("Successfully updated.");
```

Comment:
Make the hint more meaningful? e.g. `xxx has been successfully updated.`

Reply:
so what's `xxx`? project name? pom file path? Cannot figure out a proper one, what do you suggest ?